### PR TITLE
Upgrade semver checks to 0.24.1

### PR DESCRIPTION
**Pull request by rcoh**
## Checklist

- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/sdk/integration-tests/lambda/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ aws-sdk-lambda = { path = "../../build/aws-sdk/sdk/lambda" }
 aws-smithy-eventstream = { path = "../../build/aws-sdk/sdk/aws-smithy-eventstream" }
 aws-smithy-http = { path = "../../build/aws-sdk/sdk/aws-smithy-http" }
 aws-smithy-runtime = { path = "../../build/aws-sdk/sdk/aws-smithy-runtime", features = ["client", "test-util"] }
+aws-smithy-runtime-api = { path = "../../build/aws-sdk/sdk/aws-smithy-runtime-api" }
```
Comment (Velfi) [inline aws/sdk/integration-tests/lambda/Cargo.toml:18]: Why was this added?

**Review by Velfi — APPROVED** [review]

File: aws/sdk/integration-tests/lambda/Cargo.toml (hunk 1 above)
Comment (rcoh) [reply aws/sdk/integration-tests/lambda/Cargo.toml:18]: facepalm thanks

**Review by Velfi — APPROVED** [review]
